src/main/java/de/retest/recheck/review/ignore/MaxPixelDiffShouldIgnore.java

```diff
@@ -0,0 +1,99 @@
+package de.retest.recheck.review.ignore;
+
+import java.awt.Rectangle;
+import java.io.Serializable;
+import java.util.regex.MatchResult;
+import java.util.regex.Pattern;
+
+import de.retest.recheck.ignore.ShouldIgnore;
+import de.retest.recheck.review.ignore.io.RegexLoader;
+import de.retest.recheck.ui.descriptors.Element;
+import de.retest.recheck.ui.diff.AttributeDifference;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class MaxPixelDiffShouldIgnore implements ShouldIgnore {
+
+	private static final String PIXEL = "px";
+
+	private final double maxPixelDiff;
+
+	public MaxPixelDiffShouldIgnore( final double maxPixelDiff ) {
+		this.maxPixelDiff = maxPixelDiff;
+	}
+
+	@Override
+	public boolean shouldIgnoreElement( final Element element ) {
+		return false;
+	}
+
+	@Override
+	public boolean shouldIgnoreAttributeDifference( final Element element,
+			final AttributeDifference attributeDifference ) {
+		final Serializable expected = attributeDifference.getExpected();
+		final Serializable actual = attributeDifference.getActual();
+
+		if ( expected instanceof Rectangle ) {
+			return checkRectangle( (Rectangle) expected, (Rectangle) actual );
+		}
+
+		if ( expected instanceof String ) {
+			return checkString( (String) expected, (String) actual );
+		}
+
+		return false;
+	}
+
+	private boolean checkRectangle( final Rectangle expected, final Rectangle actual ) {
+		final boolean ignoreX = Math.abs( expected.x - actual.x ) <= maxPixelDiff;
+		final boolean ignoreY = Math.abs( expected.y - actual.y ) <= maxPixelDiff;
+		final boolean ignoreHeight = Math.abs( expected.height - actual.height ) <= maxPixelDiff;
+		final boolean ignoreWidth = Math.abs( expected.width - actual.width ) <= maxPixelDiff;
+		return ignoreX && ignoreY && ignoreHeight && ignoreWidth;
+	}
+
+	private boolean checkString( final String expected, final String actual ) {
+		if ( expected == null || actual == null ) {
+			return false;
+		}
+
+		if ( !expected.endsWith( PIXEL ) || !actual.endsWith( PIXEL ) ) {
+			return false;
+		}
+
+		try {
+			final double expectedDouble = Double.parseDouble( clean( expected ) );
+			final double actualDouble = Double.parseDouble( clean( actual ) );
+			return Math.abs( expectedDouble - actualDouble ) <= maxPixelDiff;
+		} catch ( final NumberFormatException e ) {
+			log.error( "Could not parse {} and {} for max pixel diff.", expected, actual, e );
+			return false;
+		}
+	}
+
+	private static String clean( final String str ) {
+		return str.replace( PIXEL, "" ).replace( ",", "." );
+	}
+
+	@Override
+	public String toString() {
+		return String.format( MaxPixelDiffShouldIgnoreLoader.FORMAT, maxPixelDiff );
+	}
+
+	public static class MaxPixelDiffShouldIgnoreLoader extends RegexLoader<MaxPixelDiffShouldIgnore> {
+
+		private static final String KEY = "maxPixelDiff=";
```

Comment:
We need to take care of the syntax. Since this can potentially ignore too much: `outline`, `margin-*`, `padding-*` and thus  potentially ignoring unexpected differences.

Reply:
What was your use-case here?

Reply:
When I thought of improving the syntax, I thought of something like:
`attribute: $key, (value|expect|actual)=$val` where
* value targets both expected or actual values
* expect targets only expected values
* actual targets only actual values
* $val is *equal* comparison

If you need something more configurable, you could do:
* `attribute: $key, (value|expect|actual): $expression` where `$expression` can be
    * `eq=$val`
    * `gt=$val`
    * ...
    * `between=$range`
    * `within=(+-)?$range`
* $range being a classical interval: `()`, `(]`, `[)` and `[]` or a direct value

We need to keep in mind that:
* `=` is a direct value
* `:` is a delegate to another matcher

Examples:
`attribute: margin-bottom, value: between=[1,2]`
`attribute: outline, expect: within=+-5`

Let's talk in the office more about that.

Reply:
>What was your use-case here?

IMHO ignoring small rendering differences among different browsers and/or target devices.

I will propose the solution later on and then we can improve it.

Also:

>We need to keep in mind that:
>
>* = is a direct value
>* : is a delegate to another matcher

This wasn't clear to me as I haven't found such information in the docs:

https://retest.github.io/docs/recheck/how-ignore-works-in-recheck/

Maybe we should add this?

Reply:
Again regarding the use case, by Kent C. Odds (JavaScript engineer at PayPal and creator of TestingJavaScript.com) via applitools.com:

>I love how Applitools Eyes has the smarts to ignore minor visual differences in your components, especially when testing on different browser types where it’s easy for layout to be off by a pixel.

I think a concise and easy way to do this is very helpful. Currently, this can only be done with a bunch of JS: https://github.com/retest/recheck-web/blob/master/.retest/recheck.ignore.js